src/compiler/parser.ts

```diff
@@ -4010,7 +4010,20 @@ module ts {
             property.name = name;
             property.questionToken = questionToken;
             property.type = parseTypeAnnotation();
-            property.initializer = allowInAnd(parseNonParameterInitializer);
+
+            // For initializers, we always want to allow 'in' expressions. For instance properties specifically,

```

Comment:
When I read this, I feel like the second sentence is going to be counter in some way to the first. It is not, and it confuses me as I expect there to be some contradiction. Maybe we can work on the wording offline.